### ISSUE TITLE
SQLSTATE[08006] [7] FATAL: password authentication failed for user

### ISSUE THREAD

**Issue by DomKrulak** [bitnami/bitnami-docker-postgresql#177]
**Description**

I've packed all required images _( no need to mention them here )_ into my `docker-compose.yml` file to run my PHP application but it couldn't connect the user into database because of a wrong password or user doesn't exist. I think it's the second. And maybe database doesn't exist.

**Steps to reproduce the issue:**

1. Place the copy of `BITNAMI_IMAGE_VERSION="12.0.0-debian-9-r22"`  into computer
3. Pack the image into `docker-compose.yml` file
4. and `docker-compose up`

**Describe the results you received:**

The respond was `SQLSTATE[08006] [7] FATAL: password authentication failed for user`. 

Than I played around for a while and striped down unnecessary packages to run just PostgreSQL with PGadmin client and connect to database using credentials defined in `docker-compose.yml` file. 

The postgresql's respond in my CLI is:
`FATAL:  password authentication failed for user "project"`
`DETAIL:  Role "project" does not exist.`
`Connection matched pg_hba.conf line 1: "host     all             all             0.0.0.0/0               md5"`

I was following the documentation and I hope I understand it well.

**Additional information you deem important (e.g. issue happens only occasionally):**

Here is my `docker-compose.yml` file

```
version: "3.7"
services:

  postgresql:
    build: ./docker/postgresql/12
    image: postgresql:12
    container_name: postgresql
    networks:
     - backend
   ports:
      - 5432
    volumes:
      - ./storage/postgresql:/bitnami/postgresql
    environment:
      POSTGRESQL_POSTGRES_PASSWORD: secret
      POSTGRESQL_USERNAME: project
      POSTGRESQL_PASSWORD: project
      POSTGRESQL_DATABASE: project
    restart: always

  pgadmin:
    image: dpage/pgadmin4
    container_name: pgadmin
    networks:
      - backend
    ports:
      - 5555:80
    depends_on:
      - postgresql
    environment:
      - PGADMIN_DEFAULT_EMAIL=admin@admin.com
      - PGADMIN_DEFAULT_PASSWORD=admin
    restart: unless-stopped

networks:
  backend:
    external: true

volumes:
  postgresql:
```

**Version**

- Output of `docker version`: 

```
2.1.0.4 (39773)
```

- Output of `docker info`:

```
Engine: 19.03.4
```

- Output of `docker-compose version` (if applicable):

```
Compose: 1.24.1
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
Docker for MAC

**Comment by DomKrulak** [comment]
It seems an issue with mounted volume and it's permissions despite setting the directory _".storage/postgresql"_ to `chmod 777 postgresql`.  Still digging deeper ...

**Comment by andresbono** [comment]
Hi @DomKrulak, maybe in order to debug your issue, you can start by doing queries to the database locally. You should be able to execute the following:

```
docker exec -it postgresql bash
```

And then, inside the container:

```
PGPASSWORD=project psql -Uproject project -c '\l'
PGPASSWORD=secret psql -Upostgres -c '\l'
```

We have reviewed the environment variables that you have set, and they seem correct.

**Comment by DomKrulak** [comment]
Hi @andresbono,

Thanks for the respond 👍 
After execution the same respond:

```
domkrulak@Dominiks-Mac-mini ~ % docker exec -it postgresql bash
I have no name!@postgresql:/$ PGPASSWORD=project psql -Uproject project -c '\l'
psql: error: could not connect to server: FATAL:  password authentication failed for user "project"
I have no name!@postgresql:/$ PGPASSWORD=secret psql -Upostgres -c '\l'
psql: error: could not connect to server: FATAL:  password authentication failed for user "postgres"
```

When using `local` volume in docker container **it works**.
But when I mount `host` volume for persistence than it goes to that failure.

I'm trying to match `uid=1001` of the running postgresql container with the host directory where I'm mounting `/bitnami/postgresql` volume for persistence but still unsuccessful. Even with `777` permissions on  `./storage/postgresql` directory.

I'm kind of lost in this. But at least I know it's a problem with mounted volume.

**Comment by andresbono** [comment]
Hi @DomKrulak, it's good to know that you have narrowed the issue to a problem with the mounted volume.

Are you cleaning the host directory before running the container so it does a first run from scratch? Does it make any difference? If you don't want to erase `./storage/postgresql`, you can just change the `docker-compose.yml` and specify a different path just for testing purposes, i.e. `./storage/postgresql_testing`.

Do the container logs show any error message?

**Comment by DomKrulak** [comment]
Hi @andresbono, 

> Are you cleaning the host directory before running the container so it does a first run from scratch? 

Yes, I always do since the lesson I've learnt when doing update to image. 

Anyway, I had to create account type "sharing only" _( I'm Mac user and probably any other user would work )_ with `UID=1001` which was assigned to "staff" group with `GID=20`. Than I created a "home" directory with current logged in user with administrative privileges  ( me ).
So I called the directory `storage` and  added `postgresql` directory. Than I mounted `/bitnami/postgresql` volume into that directory. 

Then I was able to create a connection to the `project` database.

However I've encountered some wonders during setting it. 

1. when creating directory `storage`, the owner of the directory was logged in user (not `UID=1001`) and despite that it was still able to connect. Owner of the directory `UID=501` and `GID=80`.
2. creating user group `GID=1001` and adding user `UID=1001` to that group and than creating directory `storage` with changing owner to `UID=1001` and `GID=1001`, was not able to connect.

Now I run my `storage/postgresql` directory as `UID=1001` and `GID=80`. Also good to note `GID=20` which is "staff" group works too. I've just tried these two but probably more administrative groups would work too.
